Practica3/includes/clases/Servicio.php

```diff
@@ -0,0 +1,149 @@
+            $row = $result_check->fetch_assoc();
+            $existing_id = $row['id'];
+            $existing_cantidad = $row['cantidad'];
+            $new_cantidad = $existing_cantidad + $this->cantidad;
+
+            // Invocar el método actualiza
+            return Servicio::actualiza($existing_id, $this->idPueblo, $this->idAmbito, $new_cantidad);
+        } else {
+            // No existe una entrada, insertar una nueva
+            $query_insert = "INSERT INTO servicios (idPueblo, idAmbito, cantidad) VALUES (?, ?, ?)";
+            $stmt_insert = $conn->prepare($query_insert);
+            $stmt_insert->bind_param('iii', $this->idPueblo, $this->idAmbito, $this->cantidad);
+            
+            if ($stmt_insert->execute()) {
+                $this->id = $conn->insert_id;
+                return true; // Se insertó la nueva entrada correctamente
+            } else {
+                error_log("Error BD ({$conn->errno}): {$conn->error}");
+                return false;
+            }
+        }
+    }
+
+    public static function buscaServicioPorId($idServicio)
+    {
+        $conn = Aplicacion::getInstance()->getConexionBd();
+        $query = "SELECT * FROM servicios WHERE id = ?";
+        
+        $stmt = $conn->prepare($query);
+        $stmt->bind_param('i', $idServicio);
+        if ($stmt->execute()) {
+            $result = $stmt->get_result();
+            if ($fila = $result->fetch_assoc()) {
+                return new Servicio($fila['idPueblo'], $fila['idAmbito'], $fila['cantidad'], $fila['id']);
+            }
+        }
+        return null;
+    }
+
+    public static function actualiza($id, $idPueblo, $idAmbito, $cantidad)
+    {
+        $conn = Aplicacion::getInstance()->getConexionBd();
+        $query = "UPDATE servicios SET idPueblo = ?, idAmbito = ?, cantidad = ? WHERE id = ?";
+        
+        $stmt = $conn->prepare($query);
+        $stmt->bind_param('iiii', $idPueblo, $idAmbito, $cantidad, $id);
+        
+        if ($stmt->execute()) {
+            return true;
+        } else {
+            error_log("Error BD ({$conn->errno}): {$conn->error}");
+            return false;
+        }
+    }
+
+    public static function eliminaServicioPorId($idServicio)
+    {
+        $conn = Aplicacion::getInstance()->getConexionBd();
+        $query = "DELETE FROM servicios WHERE id = ?";
+        
+        $stmt = $conn->prepare($query);
+        $stmt->bind_param('i', $idServicio);
+        
+        if ($stmt->execute()) {
+            return true;
+        } else {
+            error_log("Error BD ({$conn->errno}): {$conn->error}");
+            return false;
+        }
+    }
+
+    public static function disminuirServiciosPorEmpresaYPueblo($ambitoEmpresa, $idPueblo)
+    {
+        $conn = Aplicacion::getInstance()->getConexionBd();
+        $query = "UPDATE servicios SET cantidad = cantidad - 1 WHERE idAmbito = $ambitoEmpresa AND idPueblo = $idPueblo";
```

Comment:
aquí no utilizáis prepared statements ni utilzáis mysql_realscape_string. Hay que estar muy seguros que ésto está bien validado fuera de esta clase !.